paimon-core/src/main/java/org/apache/paimon/utils/BranchManager.java

```diff
@@ -124,11 +144,114 @@ public void deleteBranch(String branchName) {
             LOG.info(
                     String.format(
                             "Deleting the branch failed due to an exception in deleting the directory %s. Please try again.",
-                            getBranchPath(tablePath, branchName)),
+                            getBranchPath(fileIO, tablePath, branchName)),
                     e);
         }
     }
 
+    /** Replace specify branch to main branch. */
+    public void replaceBranch(String branchName) {
+        checkArgument(!StringUtils.isBlank(branchName), "Branch name '%s' is blank.", branchName);
+        checkArgument(branchExists(branchName), "Branch name '%s' not exists.", branchName);
+        try {
+            // 0. Cache previous tag,snapshot,schema directory.
+            Path tagDirectory = tagManager.tagDirectory();
+            Path snapshotDirectory = snapshotManager.snapshotDirectory();
+            Path schemaDirectory = schemaManager.schemaDirectory();
+            // 1. Calculate and copy the snapshots, tags and schemas which should be copied from the
+            // main branch to target branch.
+            calculateCopyMainBranchToTargetBranch(branchName);
+            // 2. Update the Main Branch File to the target branch.
+            updateMainBranchToTargetBranch(branchName);
+            // 3.Drop the previous main branch, including snapshots, tags and schemas.
+            dropPreviousMainBranch(tagDirectory, snapshotDirectory, schemaDirectory);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /** Calculate copy main branch to target branch. */
+    private void calculateCopyMainBranchToTargetBranch(String branchName) throws IOException {
+        TableBranch fromBranch =
+                this.branches().stream()
+                        .filter(branch -> branch.getBranchName().equals(branchName))
+                        .findFirst()
+                        .orElse(null);
+        if (fromBranch == null) {
+            throw new RuntimeException(String.format("No branches found %s", branchName));
+        }
+        // Copy tags.
+        List<TableTag> tags = tagManager.tableTags();
+        TableTag fromTag =
+                tags.stream()
+                        .filter(
+                                tableTag ->
+                                        tableTag.getTagName()
+                                                .equals(fromBranch.getCreatedFromTag()))
+                        .findFirst()
+                        .get();
+        for (TableTag tag : tags) {
+            if (tagManager.branchTagExists(branchName, tag.getTagName())) {
+                // If it already exists, skip it directly.
+                continue;
+            }
+            if (tag.getCreateTime() < fromTag.getCreateTime()) {
```

Comment:
Could the referred `snapshot-id` by tag be used directly in this filter condition?

Reply:
Thus the `TableTag` and `TagManager::branchTableTags()` can be removed, how do you think?